libr/core/project.c

```diff
@@ -673,24 +674,21 @@ R_API bool r_core_project_save(RCore *core, const char *prj_name) {
 		free (prj_bin_dir);
 		free (bin_file);
 	}
-	if (r_config_get_i (core->config, "prj.git")) {
-		char *cwd = r_sys_getdir ();
+	if (r_config_get_i (core->config, "prj.vc")) {
 		char *git_dir = r_str_newf ("%s" R_SYS_DIR ".git", prj_dir);
-		if (r_sys_chdir (prj_dir)) {
+		if (!r_config_get_i (core->config, "vc.rvc")) {
```

Comment:
use r_config_get_b as long as its bool. but... isnt this var suposed to be used to select the default version control software to be used for projects? better name it prj.vc. 

imho prj.vc="" = use r2 rvc
if prj.vc="git", then use git
prj.vc=? -> should list all the available version control systems

will be good to add mercurial probably, and shouldnt be hard to add cvs and such too.

so... what if we have an RvcPlugin struct that contains the callbacks to implement all the VC primitives instead of totally separate apis? and make the prj.vc pick the one by just replacing a pointer in one place?

Reply:
We already talked about this on Telegram
![tg](https://user-images.githubusercontent.com/60626307/115952062-5aee5680-a4e4-11eb-8e6f-1763d4fb1cfa.png) the reason it is not so easy is because radare2 treats rvc and other VCs (git for now) fundamentally differently. For example the adding APIs returns a Blobs *. this could be abstracted though but it will end up killing certain functionalities.

Reply:
ideally i would like to have the same high level primitives to work across all VC tools. But i think this PR is in a good shape for merging it and continue working on top of this